builders/xcm/fees.md

```diff
@@ -0,0 +1,230 @@
+
+In general, the process can be described as follows:
+
+1. Some assets need to be provided
+2. The exchange of assets for compute time (or weight) must be negotiated
+3. The XCM operations will be performed as instructed and any leftover assets will be sent to a destination account
+
+### XCM Instructions {: #xcm-instructions }
+
+There are a number of XCM instructions that can be used to provide assets, including the `WithdrawAsset` and `ReserveAssetDeposited` instructions. If the assets are provided via the `ReserveAssetDeposited` instruction, the assets are not always relayed from a trusted source. As such, the `ClearOrigin` instruction can be used to ensure that later instructions cannot act with the authority of the original origin.
+
+To exchange the assets for compute time, the `BuyExecution` instruction is used. Then with any remaining assets the `DepositAsset` instruction can be used to send the assets to a beneficiary.
+
+The following is an overview of each of these instructions:
+
+- **`WithdrawAsset(assets: MultiAssets)`** - removes the on-chain asset(s) and holds them in the Holding Register, a temporary position in the Cross-Consensus Virtual Machine (XCVM), which can then be used to pay for XCM execution fees
+- **`ReserveAssetDeposited(assets: MultiAssets)`** - accrues the asset(s) which have been received into the sovereign account of the local consensus system into holding
+- **`ClearOrigin`** - clears the origin to ensure that later instructions cannot command the authority of the original origin. Useful in cases where the instructions are being relayed from an untrusted source, as often is the case with `ReserveAssetDeposited`
+- **`BuyExecution(fees: MultiAsset, weight_limit: Option<Weight>)`** - takes asset(s) from holding to pay for execution fees. The amount that is actually used is determined by the interpreting system. If the fees are more than the specified amount then the instruction will result in error. You can optionally provide the amount of weight to be purchased. If the given amount of weight is lower than the estimated weight, an error will be thrown
+- **`DepositAsset(assets: MultiAssetFilter, max_assets: Compact, beneficiary: MultiLocation`** - removes asset(s) from holding to be sent to a destination account. The maximum assets defines the maximum number of unique assets to remove from holding, any remaining assets stay in holding
+
+## Calculation of Fees for Native Assets {: #calculation-of-fees-native }
+
+Substrate has introduced a weight system that determines how heavy an extrinsic is. When it comes to paying fees, users will pay a transaction fee that is proportionate to the weight of the call that is being made. One unit of weight is defined as one picosecond of execution time.
```

Comment:
Careful here also, it does not have to be proportionate, you can spend more weight but end up paying less fees because the network is less congested.